src/Components/WebAssembly/WebAssembly.Authentication/src/Models/InteractiveRequestOptions.cs

```diff
@@ -0,0 +1,192 @@
+    /// </summary>
+    /// <remarks>
+    /// The underlying provider is free to apply these parameters as it sees fit or ignore them completely. In the default
+    /// implementations the parameters will be JSON serialized using System.Text.Json and passed as a parameter to the
+    /// underlying JavaScript implementation that handles the operation details.
+    /// </remarks>
+    public bool TryAddAdditionalParameter<[DynamicallyAccessedMembers(JsonSerialized)] TValue>(string name, TValue value)
+    {
+        ArgumentNullException.ThrowIfNull(name, nameof(name));
+        AdditionalRequestParameters ??= new();
+        return AdditionalRequestParameters.TryAdd(name, value);
+    }
+
+    /// <summary>
+    /// Tries to remove an existing additional parameter.
+    /// </summary>
+    public bool TryRemoveAdditionalParameter<TValue>(string name)
+    {
+        ArgumentNullException.ThrowIfNull(name, nameof(name));
+        return AdditionalRequestParameters != null && AdditionalRequestParameters.Remove(name);
+    }
+
+    /// <summary>
+    /// Tries to retrieve an existing additional parameter.
+    /// </summary>
+    public bool TryGetAdditionalParameter<[DynamicallyAccessedMembers(JsonSerialized)] TValue>(string name, out TValue value)
+    {
+        ArgumentNullException.ThrowIfNull(name);
+
+        value = default;
+        if (AdditionalRequestParameters == null || !AdditionalRequestParameters.TryGetValue(name, out var rawValue))
+        {
+            return false;
+        }
+        if (rawValue is JsonElement json)
+        {
+            value = Deserialize(json);
+            AdditionalRequestParameters[name] = value;
+            return true;
+        }
+        else
+        {
+            value = (TValue)rawValue;
+            return true;
+        }
+
+        [UnconditionalSuppressMessageAttribute(
+            "Trimming",
+            "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+            Justification = "All the types serialized and deserialized have annotated APIs on the parent method that ensure the members are preserved.")]
+        static TValue Deserialize(JsonElement element) => element.Deserialize<TValue>();
+    }
+
+    [UnconditionalSuppressMessage(
+        "Trimming",
+        "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+        Justification = "Serializes 'InteractiveAuthenticationRequest' into a string")]
+    internal string ToState() => JsonSerializer.Serialize(this);
+
+    [UnconditionalSuppressMessage(
+        "Trimming",
+        "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+        Justification = "Deserializes InteractiveAuthenticationRequest")]
```

Comment:
That's not really a justification for why it's safe. My guess for the justification is something like "property setters are preserved by the Converter class, which in turn is preserved because [give reason]".

In fact now we come to it, is putting [JsonConverter(typeof(SomeType))] enough to preserve the members of SomeType?

Reply:
I have updated this area to make sure the members are preserved.

Reply:
I switched to use the serialization context